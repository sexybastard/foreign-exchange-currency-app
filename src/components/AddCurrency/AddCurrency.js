import React, { Component } from 'react'

const containerStyle = {
  display: 'flex',
  flex: 1,
  borderWidth: 2,
  borderStyle: 'solid',
  borderColor: 'black',
  paddingTop: 8,
  paddingBottom: 8,
  paddingLeft: 12,
  paddingRight: 12
}

const dropdownStyle = {
  display: 'flex',
  flex: 1
}

class AddCurrency extends Component {
  constructor(props) {
    super(props);
  
    this.state = {
      toggleInput: false,
      currency: 'USD'
    };
  }

  handleOnClick = () => {
    this.setState({toggleInput: !this.state.toggleInput})
  };

  handleOnChange = (event) => {
    this.setState({currency: event.target.value})
  };

  handleSubmit = () => {
    this.props.addCurrency(this.state.currency)
    this.handleOnClick();
  };

  render() {

    return (
      <React.Fragment>
      {
        this.state.toggleInput ?
        <div style={containerStyle}>
          <select style={dropdownStyle} onChange={this.handleOnChange} value={this.state.currency}>
            <option value="USD">USD</option>
            <option value="CAD">CAD</option>
            <option value="IDR">IDR</option>
            <option value="GBP">GBP</option>
            <option value="CHF">CHF</option>
            <option value="SGD">SGD</option>
            <option value="INR">INR</option>
            <option value="MYR">MYR</option>
            <option value="JPY">JPY</option>
            <option value="KRW">KRW</option>
          </select>
          <button onClick={this.handleSubmit}>Submit</button>
        </div> :
        <div style={containerStyle} onClick={this.handleOnClick}>
          <label style={{fontWeight: 'bold'}}>(+) Add More Currency</label>
        </div>
      }
      </React.Fragment>
    );
  }
}

export default AddCurrency;