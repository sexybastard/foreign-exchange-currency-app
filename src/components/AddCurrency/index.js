import { connect } from 'react-redux'
import AddCurrency from './AddCurrency'
import { addCurrency } from '../../actions'

const mapStateToProps = (state, ownProps) => ({
  ratesNameMap: state.ratesNameMap
})

const mapDispatchToProps = (dispatch, ownProps) => ({
  addCurrency: (payload) => dispatch(addCurrency(payload))
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AddCurrency)