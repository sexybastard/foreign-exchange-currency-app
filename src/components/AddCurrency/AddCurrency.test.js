import React from 'react';
import { shallow } from 'enzyme';
import AddCurrency from './AddCurrency';

describe("Component: AddCurrency", () => {
  it('AddCurrency renders correctly without data', () => {
    shallow(<AddCurrency />)
  });
  it("AddCurrency should match snapshoot", () => {
    const wrapper = shallow(<AddCurrency />)
    expect(wrapper.getElements()).toMatchSnapshot()
  });
  it('Simulates on click add more currency', () => {
    const wrapper = shallow(<AddCurrency />)
    expect(wrapper.state('toggleInput')).toBe(false)
    wrapper.find('div').first().simulate('click')
    expect(wrapper.state('toggleInput')).toBe(true)
  });
  it('Simulates on change currency click', () => {
    const wrapper = shallow(<AddCurrency />)
    expect(wrapper.state('currency')).toBe('USD')
    wrapper.find('div').first().simulate('click')
    wrapper.find('select').first().simulate('change', { target: { value: 'IDR' } })
    expect(wrapper.state('currency')).toBe('IDR')
  });
});