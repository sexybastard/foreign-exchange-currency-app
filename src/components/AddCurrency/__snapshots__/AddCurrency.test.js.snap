// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: AddCurrency AddCurrency should match snapshoot 1`] = `
Array [
  <UNDEFINED>
    <div
      onClick={[Function]}
      style={
        Object {
          "borderColor": "black",
          "borderStyle": "solid",
          "borderWidth": 2,
          "display": "flex",
          "flex": 1,
          "paddingBottom": 8,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 8,
        }
      }
    >
      <label
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        (+) Add More Currency
      </label>
    </div>
  </UNDEFINED>,
]
`;
